hw/xbox/mcpx_apu.c

```diff
@@ -1511,6 +1571,17 @@ static void se_frame(void *opaque)
         }
     }
 
+    if (d->set_irq) {
+#if USE_APU_THREAD
+        qemu_mutex_lock_iothread();
+        update_irq(d);
+        qemu_mutex_unlock_iothread();
+#else
+        update_irq(d);
+#endif
+        d->set_irq = false;
+    }
```

Comment:
Shouldn't this happen after the debug `GENERATE_MIXBIN_BEEP` to ensure (and hint for other developers) that there are valid samples when the IRQ handler runs?

Reply:
To be honest I'm not sure exactly when the hardware will assert an IRQ--we need to do some more research there. This is added here to handle stream notifications. I think whether after or before the mixbin beep doesn't matter too much as the samples stored in the temporary buffer will still be the same. I can move it down though as it looks a little cleaner.